modules/drivers/bigquery/src/metabase/driver/bigquery/query_processor.clj

```diff
@@ -351,11 +351,30 @@
   [_ value]
   (hx/cast :float64 value))
 
-
 (defmethod sql.qp/->honeysql [:bigquery :regex-match-first]
   [driver [_ arg pattern]]
   (hsql/call :regexp_extract (sql.qp/->honeysql driver arg) (sql.qp/->honeysql driver pattern)))
 
+(defn- percentile->quantile
+  [x]
+  (loop [x     (double x)
+         power (int 0)]
+    (if (zero? (- x (Math/floor x)))
+      [(Math/round x) (Math/round (Math/pow 10 power))]
+      (recur (* 10 x) (inc power)))))
+
+(defmethod sql.qp/->honeysql [:bigquery :percentile]
+  [driver [_ arg p]]
+  (let [[offset quantiles] (percentile->quantile p)]
+    (hsql/raw (format "APPROX_QUANTILES(%s, %s)[OFFSET(%s)]"
```

Comment:
Is there a reason that we went with `APPROX_QUANTILES` instead of `PERCENTILE_CONT`? BQ supports both.

Reply:
Yes. PERCENTILE_CONT in BQ is an analytic function and my understanding is those are window functions and do not play well with vanilla GROUP BYs so we'd effectively have to rewrite any query where PERCENTILE_CONT is used as a window query for all the aggregations. I'm open to doing this as a followup, but I thought it more important to get the first version through the door.

Reply:
Ahhh, cool that makes perfect sense!